src/gluonts/mx/distribution/nan_mixture.py

```diff
@@ -0,0 +1,173 @@
+from typing import Dict, List, Tuple, Optional
+
+import numpy as np
+
+# Third-party imports
+from mxnet import gluon
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.model.common import Tensor
+
+# Relative imports
+from .distribution import (
+    Distribution,
+    _expand_param,
+    _index_tensor,
+    getF,
+    nans_like,
+)
+from .distribution_output import DistributionOutput
+from .mixture import MixtureDistribution, MixtureDistributionOutput
+from .deterministic import Deterministic, DeterministicOutput
+
+
+class NanMixture(MixtureDistribution):
+    r"""
+    A mixture distribution of a NaN-valued Deterministic distribution and Distribution
+
+    Parameters
+    ----------
+    nan_prob
+        A tensor of the probabilities of missing values. The entries should all be positive
+        and smaller than 1. All axis should either coincide with the ones from the component distributions,
+        or be 1 (in which case, the NaN probability is shared across the axis).
+    distribution
+        A Distribution object representing the Distribution of non-NaN values.
+        Distributions can be of different types. Each component's support
+        should be made of tensors of shape (..., d).
+    F
+        A module that can either refer to the Symbol API or the NDArray
+        API in MXNet
+    """
+
+    is_reparameterizable = False
+
+    @validated()
+    def __init__(
+        self, nan_prob: Tensor, distribution: Distribution, F=None
+    ) -> None:
+
+        F = getF(nan_prob)
+
+        mixture_probs = F.stack(1 - nan_prob, nan_prob, axis=-1)
+        super().__init__(
+            mixture_probs=mixture_probs,
+            components=[
+                distribution,
+                Deterministic(value=nans_like(nan_prob)),
+            ],
+        )
+
+    @property
+    def distribution(self):
+        return self.components[0]
+
+    @property
+    def nan_prob(self):
+        return self.mixture_probs.slice_axis(axis=-1, begin=1, end=2).squeeze(
+            axis=-1
+        )
+
+    def log_prob(self, x: Tensor) -> Tensor:
+        F = self.F
+
+        # masking data NaN's with ones to prevent NaN gradients
+        x_non_nan = F.where(x != x, F.ones_like(x), x)
+
+        # calculate likelihood for values which are not NaN
+        non_nan_dist_log_likelihood = F.where(
+            x != x,
+            -x.ones_like() / 0.0,
+            self.components[0].log_prob(x_non_nan),
+        )
+
+        log_mix_weights = F.log(self.mixture_probs)
+
+        # stack log probabilities of components
+        component_log_likelihood = F.stack(
+            *[non_nan_dist_log_likelihood, self.components[1].log_prob(x)],
+            axis=-1,
+        )
+        # compute mixture log probability by log-sum-exp
+        summands = log_mix_weights + component_log_likelihood
+        max_val = F.max_axis(summands, axis=-1, keepdims=True)
+
+        # catch edge case max_val = -inf
```

Comment:
When is this edge case happening? In theory you should never have this unless the values of both distributions are `-inf`, right? But if the value is not nan then the proper distribution should not give `-inf`.

Reply:
I forgot to remove that, thanks! This is needed if we allow nan_prob=1, but we realized  that nan_prob has to be in (0,1) for multiple reasons. I will remove it.